src/visualizers/widgets/PlotlyGraph/PlotlyDescExtractor.js

```diff
@@ -27,6 +27,17 @@ define([], function () {
         SCATTER_POINTS_3D: 'scatter3d'
     };
 
+    const needsTightLayout = function (desc, label) {
+        let needsTightLayout = false;
+        for(let subGraph of desc.subGraphs){
+            needsTightLayout = !!subGraph.title && !!subGraph[label];
```

Comment:
This would be a little more legible if this variable was renamed to make obvious what it was checking. It seems to me that something like `hasTitleAndAxisLabel` might be more appropriate.

I personally prefer early returns to using `break` but that is a bit more of a judgement call. As an example:
```javascript
for (let subGraph of desc.subGraphs) {
    const hasTitleAndAxisLabel = !!subGraph.title && !!subGraph[label];
    if (hasTitleAndAxisLabel) {
        return true;
    }
}
return false;
```

That said, it could be even more legible (and concise) to just use `find`:
```javascript
const needsTightLayout = function (desc, label) {
    return !!desc.subGraphs.find(subGraph => hasTitleAndAxisLabel(subGraph, label);
};

const hasTitleAndAxisLabel = function (subGraph, axisLabel) {
    return subGraph.title && subGraph[axisLabel];
};
```